.github/workflows/ci.yml

```
@@ -111,26 +114,46 @@ jobs:
           fi
           echo "✓ Build successful"
           ls -lh build/rp2040-converter.*
-      
+
       - name: Install ARM toolchain for memory analysis
+        timeout-minutes: 5  # Explicit timeout for apt operations
         run: |
```

actionlint (1.7.10)

[error] shellcheck reported issue in this script: SC2086:info:14:8: Double quote to prevent globbing and word splitting

(shellcheck)